xarray/backends/zarr.py

```diff
@@ -255,8 +262,9 @@ def _extract_zarr_variable_encoding(variable, raise_on_invalid=False):
 
     return encoding
 
-# copied from conventions.encode_cf_variable
 
+# Function below is copied from conventions.encode_cf_variable.
+# The only change is to raise an error for object dtypes.
```

Comment:
netCDF files can't store object dtypes, either -- that's what `ensure_dtype_not_object` is for. But maybe that's no enough here for some reason?

Reply:
The problem is that the way `ensure_dtype_not_object` was encoding the objects (usually multi-length strings) was not round-tripping properly with zarr. For now, raising an error seems like an ok short-term solution.  I hope it will be much easier to fix this encoding stuff after your backend / encoding refactor is complete.

Is it ok to leave this as is for now?